fanuc_r2000ib_support/urdf/r2000ib210f_macro.xacro

```diff
@@ -0,0 +1,182 @@
+        <geometry>
+          <mesh filename="package://fanuc_r2000ib_support/meshes/r2000ib210f/collision/link_6.stl"/>
+        </geometry>
+      </collision>
+    </link>
+
+    <!-- joints: main serial chain -->
+    <joint name="${prefix}joint_1" type="revolute">
+      <origin xyz="0 0 0.670" rpy="0 0 0"/>
+      <parent link="${prefix}base_link"/>
+      <child link="${prefix}link_1"/>
+      <axis xyz="0 0 1"/>
+      <limit lower="${radians(-180)}" upper="${radians(180)}" effort="0" velocity="${radians(95)}"/>
+    </joint>
+    <joint name="${prefix}joint_2" type="revolute">
+      <origin xyz="0.312 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_1"/>
+      <child link="${prefix}link_2"/>
+      <axis xyz="0 1 0"/>
+      <limit lower="${radians(-60)}" upper="${radians(76)}" effort="0" velocity="${radians(90)}"/>
+    </joint>
+    <joint name="${prefix}joint_3" type="revolute">
+      <origin xyz="0 0 1.075" rpy="0 0 0"/>
+      <parent link="${prefix}link_2"/>
+      <child link="${prefix}link_3"/>
+      <axis xyz="0 -1 0"/>
+      <limit lower="${radians(-132)}" upper="${radians(230)}" effort="0" velocity="${radians(95)}"/>
+    </joint>
+    <joint name="${prefix}joint_4" type="revolute">
+      <origin xyz="0 0 0.225" rpy="0 0 0"/>
+      <parent link="${prefix}link_3"/>
+      <child link="${prefix}link_4"/>
+      <axis xyz="-1 0 0"/>
+      <limit lower="${radians(-360)}" upper="${radians(360)}" effort="0" velocity="${radians(120)}"/>
+    </joint>
+    <joint name="${prefix}joint_5" type="revolute">
+      <origin xyz="1.280 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_4"/>
+      <child link="${prefix}link_5"/>
+      <axis xyz="0 -1 0"/>
+      <limit lower="${radians(-125)}" upper="${radians(125)}" effort="0" velocity="${radians(120)}"/>
+    </joint>
+    <joint name="${prefix}joint_6" type="revolute">
+      <origin xyz="0.235 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_5"/>
+      <child link="${prefix}link_6"/>
+      <axis xyz="0 0 -1"/>
+      <limit lower="${radians(-360)}" upper="${radians(360)}" effort="0" velocity="${radians(190)}"/>
+    </joint>
+
+    <!-- ROS-Industrial 'base' frame: base_link to Fanuc World Coordinates transform -->
+    <link name="${prefix}base" />
+    <joint name="${prefix}base_link-base" type="fixed">
+      <origin xyz="0 0 0.670" rpy="0 0 0"/>
+      <parent link="${prefix}base_link"/>
+      <child link="${prefix}base"/>
+    </joint>
+
+    <!-- ROS-Industrial 'flange' frame: attachment point for EEF models -->
+    <link name="${prefix}flange" />
+    <joint name="${prefix}joint_6-flange" type="fixed">
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <parent link="${prefix}link_6" />
+      <child link="${prefix}flange" />
+    </joint>
+
+    <!-- ROS-Industrial 'tool0' frame: all-zeros tool frame -->
+    <link name="${prefix}tool0" />
+    <joint name="${prefix}link_6-tool0" type="fixed">
+      <origin xyz="0 0 0" rpy="${pi} ${-pi/2.0} 0" />
```

Comment:
I've changed `tool0`'s orientation to match other Fanuc robots.

@demorise: you have `link_6` rotated `pi` over Y+. Did you do this for a specific reason? We prefer to not diverge from [REP-103](https://www.ros.org/reps/rep-0103.html#coordinate-frame-conventions) for link-local frames (such as `link_6`) and "rectify" the situation with `tool0` or additional toolframes.

Reply:
It looks like the CAD file origin is oriented different from other fanuc robots, so without the rotation, I get something like this:
![r2000ib_link6_issue](https://user-images.githubusercontent.com/22728175/88975034-2b752b00-d27f-11ea-87dc-a8eaf2d27a02.png)

Reply:
I'm pretty sure I re-oriented all meshes to follow REP-103, at least for the link-local frames.

I'll check again, but in the meantime: could you make sure you're using the files from this PR?

Reply:
I used the files from this PR. My original meshes have a lot of transparent sections.

Reply:
So this had nothing to do with mesh orientation.

`joint_6` just had an incorrect axis specification. I've corrected it in 5722c8c.

Thanks for spotting that btw, as I'd assumed the axis specs were OK (as it's something which is really easy to notice).

> My original meshes have a lot of transparent sections.

That's the problem with the normals I referenced in 7321ccd.

Reply:
I'm going to resolve this as I believe I understand why you had that rotation (and why it's no longer needed).

Reply:
> That's the problem with the normals I referenced in [7321ccd](https://github.com/ros-industrial/fanuc_experimental/commit/7321ccd3029fa1b59c5c737926280b34fb5f8d51).
Correct. I mentioned that as a confirmation that I was using the correct files. If you don't mind, I'll like to know how you fixed the issue with the normals for future reference.

Reply:
Meshlab.

Use `filters->Normals, Curvatures and .. ->Re-orient all faces coherently`, then if needed `Invert faces orientation` (from the same menu).

Otherwise I use Blender.

Reply:
Thanks. ...and thanks for the effort you've put into this. I just tried the latest commit and it works fine now.
![Screenshot from 2020-07-31 10-00-17](https://user-images.githubusercontent.com/22728175/89048128-ffa48480-d314-11ea-9cff-d3427f1aa45a.png)

Reply:
Thanks.

Could you take a look at https://github.com/ros-industrial/fanuc_experimental/pull/57#issuecomment-667141786?